src/main/java/winstone/HostConfiguration.java

```diff
@@ -98,11 +93,12 @@ public HostConfiguration(Server server, String hostname, ClassLoader commonLibCL
                     }
                     String extension = mapping.substring(0, delimPos);
                     String mimeType = mapping.substring(delimPos + 1);
-                    this.mimeTypes.addMimeMapping(extension.toLowerCase(), mimeType);
+                    server.getMimeTypes().addMimeMapping(extension.toLowerCase(), mimeType);
                 }
             }
         }
-
+        server.setRequestLog(configureAccessLog("webapp"));
```

Comment:
`configureAccessLog` can return null, so I think we should wrap this in a null check:

```suggestion
        RequestLog requestLog = configureAccessLog("webapp");
        if (requestLog != null) {
            server.setRequestLog(requestLog);
        }
```